### ISSUE TITLE
Set tuning indices

### ISSUE THREAD

**Issue by pecto2020** [topepo/caret#1235]
I'm performing a nested RFE with ranger applied to a binary classification task. Data are spatially distributed and good practice in my case is to group them by location (several records share the same coordinates).
Imagine I have a 4 folds outer loop for evaluation of subset size and a 3 folds inner loop for hp tuning.
The indices for the outer 4 folds is a list of 4 vectors with indices related to each fold.
I can pass a list of indices to the rfeControl function, it nicely works.

When it comes to the inner loop (3 folds), the indices file is a list of 4 elements (outer folds) with each element containing 3 inner vectors. So I have a nested list that needs to be passed to the trainControl function. Unfortunately, I got this error.

Error in { : task 1 failed - "`index` should be lists of integers."

Is there a way to solve this issue? It's very important to have a grouped hyperparameter tuning because if data points sharing in the same location are either in the training and in the test set I run into data leakage. 

Thanks
